### PR TITLE
[SUS-415] Automatically include all i18n files in LocalisationCache rebuild script. 

### PR DESCRIPTION
**Pull request by ArturKlajnerok**
We should include all localisation files (i18n & aliases) in LocalisationCache rebuild process, this will alow us to build cache file that contains all the messages keys and we will be able to remove manual includes like here :  https://github.com/Wikia/config/blob/dev/CommonExtensions.php#L122

https://wikia-inc.atlassian.net/browse/SUS-415

/cc @mixth-sense

### CODE REVIEW

File: maintenance/rebuildLocalisationCache.php [hunk 1]
```diff
@@ -151,6 +158,28 @@ function ( $item ) use ( $firstCodes ) {
 	}
 
 	/**
+	 * Helper function to include all localisation files to localisation cache rebuild process.
+	 * It looks for i18n and alias files from whole code directory and registers them in global array.
+	 */
+	private function includeAllMessagesFiles() {
+		global $wgExtensionMessagesFiles;
+
+		$directory = new RecursiveDirectoryIterator('/usr/wikia/source/app/');
```
Comment (ghost) [inline maintenance/rebuildLocalisationCache.php]: Please, do not hardcode paths. Use `$IP`.

File: maintenance/rebuildLocalisationCache.php [hunk 2]
```diff
@@ -151,6 +158,28 @@ function ( $item ) use ( $firstCodes ) {
...
+		$iterator = new RecursiveIteratorIterator($directory);
+		$files = new RegexIterator($iterator, self::LOCALISATION_FILE_REGEX, RecursiveRegexIterator::GET_MATCH);
+
+		foreach ($files as $filepath) {
+			if (!isset($filepath[0])) continue;
```
Comment (ghost) [inline maintenance/rebuildLocalisationCache.php]: When does it happen?

File: maintenance/rebuildLocalisationCache.php [hunk 3]
```diff
@@ -151,6 +158,28 @@ function ( $item ) use ( $firstCodes ) {
...
+
+			if (strpos($filepath[0], self::I18N_FILE_EXTENSION) !== false ) {
+				$wgExtensionMessagesFiles[basename($filepath[0], self::I18N_FILE_EXTENSION)] = $filepath[0];
```
Comment (ghost) [inline maintenance/rebuildLocalisationCache.php]: How about `$key = basename(...);` and a single `$wgExtensionMessagesFiles[$key]` assignment?

File: maintenance/rebuildLocalisationCache.php (hunk 1 above)
Comment (ArturKlajnerok) [inline maintenance/rebuildLocalisationCache.php]: Used `$IP` in : a85c80c

File: maintenance/rebuildLocalisationCache.php (hunk 3 above)
Comment (ArturKlajnerok) [inline maintenance/rebuildLocalisationCache.php]: Good catch, `$key` it is :  5025a76

File: maintenance/rebuildLocalisationCache.php (hunk 2 above)
Comment (ArturKlajnerok) [inline maintenance/rebuildLocalisationCache.php]: It should not happen, as if file exists and is passed to loop, there will always be `[0]` that contains full path and `[1]` the substring that matches to regex. Changed in : 725ee7e

File: maintenance/rebuildLocalisationCache.php [hunk 4]
```diff
@@ -151,6 +158,28 @@ function ( $item ) use ( $firstCodes ) {
 	}
 
 	/**
+	 * Helper function to include all localisation files to localisation cache rebuild process.
+	 * It looks for i18n and alias files from whole code directory and registers them in global array.
+	 */
+	private function includeAllMessagesFiles() {
+		global $wgExtensionMessagesFiles;
```
Comment (ghost) [inline maintenance/rebuildLocalisationCache.php]: It is perfectly fine to use `global $wgExtensionMessagesFiles, $IP;` which we normally do everywhere else in the codebase.

File: maintenance/rebuildLocalisationCache.php [hunk 5]
```diff
@@ -151,6 +158,28 @@ function ( $item ) use ( $firstCodes ) {
 	}
 
 	/**
+	 * Helper function to include all localisation files to localisation cache rebuild process.
+	 * It looks for i18n and alias files from whole code directory and registers them in global array.
+	 */
+	private function includeAllMessagesFiles() {
```
Comment (ghost) [inline maintenance/rebuildLocalisationCache.php:164]: Make sure, the method follows MW coding conventions. Namely, it's all about spaces after ('s and before )'s, as they are present in the above `execute()` methods.

File: maintenance/rebuildLocalisationCache.php [hunk 6]
```diff
@@ -31,6 +31,11 @@
 require_once( dirname( __FILE__ ) . '/Maintenance.php' );
```
Comment (ghost) [inline maintenance/rebuildLocalisationCache.php]: Let's use `__DIR__` here.

**Comment by ghost** [comment]
@ArturKlajnerok - just a few cosmetic changes, otherwise it LGTM. :+1:

File: maintenance/rebuildLocalisationCache.php (hunk 5 above)
Comment (ArturKlajnerok) [inline maintenance/rebuildLocalisationCache.php:164]: As far I've checked there's now space between `(` and `)` if there's no params in method.

File: maintenance/rebuildLocalisationCache.php (hunk 6 above)
Comment (ArturKlajnerok) [inline maintenance/rebuildLocalisationCache.php]: Changed in : 351ca0b

File: maintenance/rebuildLocalisationCache.php (hunk 4 above)
Comment (ArturKlajnerok) [inline maintenance/rebuildLocalisationCache.php]: Changed in : 351ca0b

**Comment by ghost** [comment]
Excellent, :+1:

**Pull request merged**